lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -11,6 +11,10 @@ def initialize(catalog, relationship_graph, prioritizer)
     @prioritizer = prioritizer
   end
 
+  def relationship_graph=(relationship_graph)
```

Comment:
Why not add attr_writer for :relationship_graph instead of making your own setter?